cmd/journald.go

```diff
@@ -0,0 +1,159 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/go-logr/zapr"
+	"github.com/metal-toolbox/auditevent"
+	"github.com/metal-toolbox/auditevent/helpers"
+	"github.com/spf13/cobra"
+	"go.uber.org/zap"
+	"golang.org/x/sync/errgroup"
+
+	"github.com/metal-toolbox/audito-maldito/ingesters/journald"
+	"github.com/metal-toolbox/audito-maldito/internal/common"
+	"github.com/metal-toolbox/audito-maldito/internal/health"
+	"github.com/metal-toolbox/audito-maldito/internal/metrics"
+	"github.com/metal-toolbox/audito-maldito/internal/util"
+	"github.com/metal-toolbox/audito-maldito/processors/auditd"
+	"github.com/metal-toolbox/audito-maldito/processors/auditd/dirreader"
+	"github.com/metal-toolbox/audito-maldito/processors/sshd"
+)
+
+var journaldCmd = &cobra.Command{
+	Use:   "journald",
+	Short: "Uses coreos/go-systemd code to access journald for data ingestion.",
+	Long: `Uses coreos/go-systemd code to access journald for data ingestion.
+	 Processes sshd logs and audit events.`,
+	Run: func(cmd *cobra.Command, args []string) {
+		ctx, stop := signal.NotifyContext(context.Background(), os.Interrupt, syscall.SIGTERM)
```

Comment:
use `cmd.Context()` instead of `context.Background()`, that'll capture the context that cobra actually initiates.

Reply:
done